app/src/ui/dispatcher/error-handlers.ts

```diff
@@ -437,7 +445,20 @@ export async function localChangesOverwrittenHandler(
     return error
   }
 
-  dispatcher.recordErrorWhenSwitchingBranchesWithUncommmittedChanges()
+  // This indicates to us whether the action which triggered the
+  // LocalChangesOverwritten was the AppStore _checkoutBranch method.
+  // Other actions that might trigger this error such as deleting
+  // a branch will not provide this specific gitContext and that's
+  // how we know we can safely move the changes to the destination
+  // branch.
+  if (gitContext === undefined || gitContext.kind !== 'checkout') {
+    dispatcher.recordErrorWhenSwitchingBranchesWithUncommmittedChanges()
+    return error
+  }
 
-  return error
+  const { branchToCheckout } = gitContext
+
+  await dispatcher.moveChangesToBranchAndCheckout(repository, branchToCheckout)
```

Comment:
i just noticed this `await` in another read through the code. i think we should not use `await` here so that we don't have a second `_checkoutBranch` executed before the first `_checkoutBranch` can finish. however this requires us to exit from the initial `_checkoutBranch` early if `checkout` is not successful. that feels like a more risky change to the behavior of `checkout`, which prior to now has never checked wether `checkout` was actually successful before making some state updates/refreshes.

since it doesn't seem like this is causing any obvious bugs i'm going to say that this is fine for now, but is something we should keep an eye on. 👀